shef_loader/cda_loader.py

```diff
@@ -0,0 +1,393 @@
+            self._logger.debug(f"time_series: {self._time_series}")
+        if self._time_series:
+            time_series: list[CdaValue] = []
+            for ts in self._time_series:
+                time = self.get_unix_timestamp(ts[0])
+                time_series.append(CdaValue(time, ts[1], 0))
+            post_data: TimeseriesPayload = {
+                "name": self.get_time_series_name(sv),
+                "office-id": "LRL",
+                "units": self.transform.units,
+                "values": time_series,
+            }
+            match_index = self.find_matching_payload_index(post_data)
+            if not match_index:
+                self._payloads.append(post_data)
+            else:
+                match_payload = self._payloads[match_index]
+                match_payload["values"].append(*time_series)
+        self._time_series = []
+
+    def create_write_task(self, post_data: TimeseriesPayload) -> Coroutine:
+        """
+        Create an async CDA POST request coroutine for provided post_data
+        """
+        return asyncio.to_thread(
+            self._cwms.write_ts, post_data, store_rule="REPLACE WITH NON MISSING"
+        )
+
+    async def process_write_tasks(self) -> None:
+        """
+        Submit CDA POST requests and report results
+        """
+        if self._logger:
+            self._logger.info("Beginning CWMS-Data-API POST tasks...")
+        start_time = time.time()
+        for response_future in asyncio.as_completed(self._write_tasks):
+            response: Response = await response_future
+            if not response.request.body:
+                raise shared.LoaderException("Error retrieving request payload")
+            payload: TimeseriesPayload = json.loads(response.request.body)
+            tsid = payload["name"]
+            value_count = len(payload["values"])
+            if response.status_code >= 400:
+                self.record_error(
+                    tsid, value_count, response.status_code, response.content
+                )
+            else:
+                self._value_count += value_count
+                self._time_series_count += 1
+                if self._logger:
+                    self._logger.info(f"Posted {value_count} values to {tsid}")
+        process_time = time.time() - start_time
+        if self._logger:
+            self._logger.info(
+                f"CWMS-Data-API POST tasks complete ({process_time:.2f} seconds)"
+            )
+
+    def record_error(
+        self, tsid: str, value_count: int, status: int, content: bytes
+    ) -> None:
+        """
+        Update error count and output message for failed CDA POST request
+        """
+        self._value_error_count += value_count
+        self._time_series_error_count += 1
+        if self._logger:
+            self._logger.error(f"HTTP {status}: {tsid} - {content.decode()}")
+
+    def find_matching_payload_index(self, payload: TimeseriesPayload) -> int | None:
```

Comment:
Change to Union[int, None]

Reply:
Fixed in [8df6c69](https://github.com/HydrologicEngineeringCenter/SHEF_processing/pull/9/commits/8df6c697d07a19b6ab781208b1093360c09d13bb)